### ISSUE TITLE
Bootstrap fails on CentOS 7

### ISSUE THREAD

**Issue by lietu** [saltstack/salt-bootstrap#513]
I just tried to upgrade my Vagrant box previously using CentOS 6.x to CentOS 7 to see what kind of issues I'd bump into.

First issue I found was that the Salt bootstrap just completely fails.

```
==> dev: Running provisioner: salt...
Checking if salt-minion is installed
salt-minion was not found.
Checking if salt-call is installed
salt-call was not found.
Bootstrapping Salt... (this may take a while)
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

/tmp/bootstrap_salt.sh  stable

Stdout from the command:

 *  INFO:  sh -- Version 2014.10.30

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   3.10.0-123.el7.x86_64
 *  INFO:   Distribution: CentOS 7.0

 *  INFO: Installing minion
 *  INFO: Found function install_centos_stable_deps
 *  INFO: Found function install_centos_stable
 *  INFO: Found function install_centos_stable_post
 *  INFO: Found function install_centos_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Found function install_centos_check_services
 *  INFO: Running install_centos_stable_deps()
curl: (22) The requested URL returned error: 404 Not Found
error: skipping http://download.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-2.noarch.rpm - transfer failed
Retrieving http://download.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-2.noarch.rpm
 * ERROR: Failed to run install_centos_stable_deps()!!!


Stderr from the command:

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  183k  100  183k    0     0  27011      0  0:00:06  0:00:06 --:--:-- 43214
```

To me it looks like it's failing to install EPEL repos correctly, however it seems the repo does exist for CentOS 7: https://fedoraproject.org/wiki/EPEL

**Comment by lietu** [comment]
```
[vagrant@dev ~]$ sudo find /etc/ -name '*-release' -print -exec cat {} \;
/etc/centos-release
CentOS Linux release 7.0.1406 (Core)
/etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

/etc/redhat-release
CentOS Linux release 7.0.1406 (Core)
/etc/system-release
CentOS Linux release 7.0.1406 (Core)
[vagrant@dev ~]$ which lsb_release && lsb_release -a
/usr/bin/which: no lsb_release in (/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/vagrant/.local/bin:/home/vagrant/bin)
```

**Comment by lietu** [comment]
Ah, sorry, seems to work when I add to my Vagrantfile:

```
config.install_type = "git"
config.install_args = "develop"
```

Though I wouldn't want to leave that there. Can anyone guestimate how long it takes for this to get to the stable version?

**Comment by MrCitron** [comment]
today http://download.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-2.noarch.rpm is 404
to correct this, we have modified our bootstrap-salt.sh file to refer to http://download.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm instead

**Comment by lietu** [comment]
Cool, thanks.

**Comment by MrCitron** [comment]
@lietu don't close this issue, it still needs to be fixed in salt code, wheter by modifying the url, or by hardening the script (using some kind of loop or querying the parent folder first to find the latest rpm ?)

**Comment by lietu** [comment]
@MrCitron fair enough, I thought the issue was 100% resolved in develop ;)

**Comment by s0undt3ch** [comment]
This is now fixed as v2014.12.11 was released as the latest bootstrap script release.

Updates to Salt:
https://github.com/saltstack/salt/pull/18929
https://github.com/saltstack/salt/pull/18930
